src/main/java/io/appform/dropwizard/actors/actor/UnmanagedBaseActor.java

```diff
@@ -56,116 +48,147 @@ public class UnmanagedBaseActor<Message> {
+                              final MessageHandlingFunction<Message, Boolean> handlerFunction,
+                              final MessageHandlingFunction<Message, Boolean> sidelineProcessorHandleFunction,
+                              final MessageHandlingFunction<Message, Boolean> expiredMessageHandlingFunction,
+                              final MessageHandlingFunction<Message, Boolean> sidelineProcessorExpiredMessageHandlingFunction,
+                              final Function<Throwable, Boolean> errorCheckFunction) {
+        this(name, config, connectionRegistry, mapper, new RandomShardIdCalculator<>(config), retryStrategyFactory,
+                exceptionHandlingFactory, clazz, handlerFunction, sidelineProcessorHandleFunction,
+                expiredMessageHandlingFunction, sidelineProcessorExpiredMessageHandlingFunction, errorCheckFunction);
+    }
+
+    public UnmanagedBaseActor(final String name,
+                              final ActorConfig config,
+                              final ConnectionRegistry connectionRegistry,
+                              final ObjectMapper mapper,
+                              final ShardIdCalculator<Message> shardIdCalculator,
+                              final RetryStrategyFactory retryStrategyFactory,
+                              final ExceptionHandlingFactory exceptionHandlingFactory,
+                              final Class<? extends Message> clazz,
+                              final MessageHandlingFunction<Message, Boolean> handlerFunction,
+                              final MessageHandlingFunction<Message, Boolean> sidelineProcessorHandleFunction,
+                              final MessageHandlingFunction<Message, Boolean> expiredMessageHandlingFunction,
+                              final MessageHandlingFunction<Message, Boolean> sidelineProcessorExpiredMessageHandlingFunction,
+                              final Function<Throwable, Boolean> errorCheckFunction) {
+
+        this(name, config, connectionRegistry.createOrGet(NamingUtils.producerConnectionName(config.getProducer())),
+                connectionRegistry.createOrGet(NamingUtils.consumerConnectionName(config.getConsumer())),
+                connectionRegistry.createOrGet(
+                        NamingUtils.sidelineProcessorConnectionName(config.getSidelineProcessor())), mapper,
+                shardIdCalculator, retryStrategyFactory, exceptionHandlingFactory, clazz, handlerFunction,
+                sidelineProcessorHandleFunction, expiredMessageHandlingFunction,
+                sidelineProcessorExpiredMessageHandlingFunction, errorCheckFunction);
+    }
+
+    public UnmanagedBaseActor(final String name,
+                              final ActorConfig config,
+                              final RMQConnection producerConnection,
+                              final RMQConnection consumerConnection,
+                              final RMQConnection sidelineProcessorConnection,
```

Comment:
Connection for sideline processor and main consumer can remain same. Consumers are never throttled due to Flow Control (only producers are throttled)